src/Host/Broker/Impl/Sessions/SessionManager.cs

```diff
@@ -59,6 +60,41 @@ ILogger<Process> hostOutputLogger
             }
         }
 
+        public void CloseAndBlockSessionsCreationForUser(IIdentity user) {
+            lock (_sessions) {
+                lock (_blockedUsers) {
+                    var userSessions = GetOrCreateSessionList(user);
+                    var sessions = userSessions.ToArray();
+                    foreach (var session in sessions) {
+                        try {
+                            userSessions.Remove(session);
+                            Task.Run(() => session.KillHost()).DoNotWait();
+                        } catch (Exception) { }
+
+                        session.State = SessionState.Terminated;
+                    }
+
+                    if (!_blockedUsers.Contains(user.Name)) {
```

Comment:
You don't need to check, it's always safe to add to a hashset (`Add` will return `false` if element was already there).